File: features/permissionResolver/src/main/java/com/walhalla/permissionresolver/permission/IOUtils.java
package com.walhalla.permissionresolver.permission;

import android.annotation.SuppressLint;
import android.os.Build;


public class IOUtils {

    @SuppressLint("ObsoleteSdkInt")
    public static boolean hasGingerbread() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.GINGERBREAD;
    }
    public static boolean hasLolipop() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
    }
    public static boolean hasMarsallow23() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M;
    }
}

File: features/permissionResolver/src/main/java/com/walhalla/permissionresolver/permission/IOUtils.kt
package com.walhalla.permissionresolver.permission

import android.annotation.SuppressLint
import android.os.Build

object IOUtils {
    @SuppressLint("ObsoleteSdkInt")
    fun hasGingerbread(): Boolean {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.GINGERBREAD
    }
    @JvmStatic
    fun hasLolipop(): Boolean {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP
    }

    @JvmStatic
    fun hasMarsallow23(): Boolean {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
    }
}
